daemons/ipa-sam/ipa_sam.c

```diff
@@ -3554,6 +3554,77 @@ static NTSTATUS getsam_interdom_trust_account(struct pdb_methods *methods,
 	return status;
 }
 
+static NTSTATUS ipasam_getsampwsid(struct pdb_methods *methods,
+				    struct samu *user,
+				    const struct dom_sid *sid)
+{
+	struct ipasam_private *ipasam_state =
+			talloc_get_type_abort(methods->private_data, struct ipasam_private);
+	TALLOC_CTX *tmp_ctx;
+	NTSTATUS status;
+	char *filter = NULL;
+	char *sid_str = NULL;
+	LDAPMessage *result = NULL;
+	LDAPMessage *entry = NULL;
+	int ret;
+	int count;
+
+	tmp_ctx = talloc_new(NULL);
+	if (tmp_ctx == NULL) {
+		return NT_STATUS_NO_MEMORY;
+	}
+
+	sid_str = sid_talloc_string(ipasam_state->idmap_ctx, tmp_ctx, sid);
+	if (sid_str == NULL) {
+		return NT_STATUS_NO_MEMORY;
```

Comment:
This will leak tmp_ctx. It should set status and goto done so that talloc_free(tmp_ctx) can be called.

Reply:
Noted.